pkg/image/containerdregistry/new.go

```diff
@@ -0,0 +1,618 @@
+package containerdregistry
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go"
+	ocispecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/operator-framework/operator-registry/pkg/image"
+)
+
+const (
+	emptyBaseImage   = "scratch"
+	rootFSTypeLayers = "layers"
+	ociSchemaVersion = 2
+	whPrefix         = ".wh."
+	whOpaque         = ".wh..wh..opq"
+)
+
+// NewImage creates a new image from the provided base image and the given tag.
+func (r *Registry) NewImage(ctx context.Context, from, imageTag image.Reference, opts ...BuildOpt) error {
```

Comment:
Pack requires a directory, but newImage doesn't need to have any content in it. You can build an image with no filesystem with this, so it could serve as a base image for holding the labels, volumes and other config.